File: algorithms/R/Reverse Integer/Reverse Integer.js
var reverse = function(x) {
    let val = Math.abs(x)
    let res = 0
    while(val !=0){
         res = (res*10) + val %10
         val = Math.floor(val/10)
    }
    if(x < 0)res = 0 - res
    return (res > ((2**31)-1) || res < (-2)**31) ? 0 : res 
};

File: scripts/algorithms/R/Reverse Integer/Reverse Integer.py
import bisect
class Solution:
    def reverse(self, x: int) -> int:

        flag = 0
        if x<0:
            x = abs(x)
            flag = 1
            
        l = [i for i in str(x)]
        l.reverse()
        
        ret = ''.join(l)
        ret = int(ret)
        
        if flag == 1:
            ret = ret*-1
            
        if ((ret >= (-(2**31))) and (ret<=((2**31)-1))):
            return ret
        else:
            return 0
